### PR TITLE
Make it compatible with OTP 24

### PR DESCRIPTION
**Pull request by hidnasio**
## Description of the change

> Update the library to make it work with OTP 24

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change
- [x] Technical Debt
- [ ] Documentation

## Related tickets

> https://app.shortcut.com/active-prospect/story/29781/upgrade-trustedform-api-to-make-it-compatible-with-otp-24

### Development and Testing

- [ ]  Lint rules pass locally.
- [ ]  The code changed/added as part of this pull request has been covered with tests, or this PR does not alter production code.
- [ ]  All tests related to the changed code pass in development, or tests are not applicable.

### Code Review

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ]  At least two engineers have been added as "Reviewers" on the pull request.
- [ ]  Changes have been reviewed by at least two other engineers who did not write the code.
- [ ]  This branch has been rebased off master to be current.

### Tracking 
- [ ]  Issue from Clubhouse has a link to this pull request.
- [ ]  This PR has a link to the issue in Clubhouse.

### QA
- [ ]  This branch has been deployed to staging and tested.

### CODE REVIEW

File: lib/mongo.ex [hunk 1]
```diff
@@ -344,7 +349,7 @@ defmodule Mongo do
   @doc """
   Similar to `count_documents/4` but unwraps the result and raises on error.
   """
-  @spec count_documents!(GenServer.server, collection, BSON.document, Keyword.t) :: result!(non_neg_integer)
+  # @spec count_documents!(GenServer.server, collection, BSON.document, Keyword.t) :: result!(non_neg_integer)
```
Comment (samullen) [inline lib/mongo.ex:352]: Do we need this uncommented?

File: lib/mongo.ex [hunk 2]
```diff
@@ -917,15 +922,6 @@ defmodule Mongo do
       opts: opts}
   end
 
-  defp singly_cursor(conn, coll, query, select, opts) do
```
Comment (samullen) [inline lib/mongo.ex:920]: Thank you for deleting this, if only because it has a stupid name.

File: lib/mongo.ex (hunk 1 above)
Comment (hidnasio) [reply lib/mongo.ex:352]: Uncommenting this line makes dialyzer to fail, in the original library they fixed that by commenting out dialyzer in the CI, I prefer to comment this line 😄

File: lib/mongo.ex (hunk 2 above)
Comment (hidnasio) [reply lib/mongo.ex:920]: It was throwing a no-used compilation warning

**Review by samullen — APPROVED** [review]

**Review by stwf — APPROVED** [review]
Why is it that we need our own version of mongodb?

**Comment by hidnasio** [comment]
@stwf The library doesn't support mongo 3.0 that is the version we are using on prod.

File: .travis.yml [hunk 3]
```diff
@@ -1,11 +1,20 @@
 sudo: required
 language: elixir
+otp_release:
```
Comment (bmartin2015) [inline .travis.yml:3]: Should this also get moved to GitHub actions?

File: mix.exs [hunk 4]
```diff
@@ -7,7 +7,7 @@ defmodule Mongodb.Mixfile do
     [app: :mongodb,
      version: @version,
      elixirc_paths: elixirc_paths(Mix.env),
-     elixir: "~> 1.4",
+     elixir: "~> 1.12",
```
Comment (bmartin2015) [inline mix.exs:10]: 😲

**Review by bmartin2015 — APPROVED** [review]

**Pull request merged**